htlcswitch/link.go

```diff
@@ -2796,26 +2774,14 @@ func (l *channelLink) HandleChannelUpdate(message lnwire.Message) {
 	l.mailBox.AddMessage(message)
 }
 
-// ShutdownIfChannelClean triggers a link shutdown if the channel is in a clean
+// ShutdownHtlcManager triggers a link shutdown if the channel is in a clean
 // state and errors if the channel has lingering updates.
 //
 // NOTE: Part of the ChannelUpdateHandler interface.
-func (l *channelLink) ShutdownIfChannelClean() error {
-	errChan := make(chan error, 1)
-
+func (l *channelLink) ShutdownHtlcManager() {
 	select {
-	case l.shutdownRequest <- &shutdownReq{
-		err: errChan,
-	}:
+	case l.shutdownRequest <- struct{}{}:
 	case <-l.quit:
-		return ErrLinkShuttingDown
-	}
-
-	select {
-	case err := <-errChan:
```

Comment:
See comments above (I think we can remove this method all together), but in the past this was sync, and now it's async.

Reply:
It appears so